src/swapRouter.ts

```diff
@@ -469,6 +467,22 @@ export abstract class SwapRouter {
     }
   }
 
+  // if slippage is very high, there's a chance of hitting max/min prices resulting in a partial fill of the swap
+  private static riskOfPartialFill(trades: AnyTradeType): boolean {
+    const FIFTY_PERCENT = new Percent(JSBI.BigInt(50), JSBI.BigInt(100))
+
+    if (trades instanceof Trade || trades instanceof V3Trade) {
+      return trades.priceImpact.greaterThan(FIFTY_PERCENT)
+    } else if (Array.isArray(trades)) {
+      for (const trade of trades) {
+        if (trade instanceof V3Trade && trade.priceImpact.greaterThan(FIFTY_PERCENT)) {
+          return true
+        }
+      }
+    }
+    return false
+  }
```

Comment:
We need to check tokenIn is native eth too

Could you also add a test that checks the calldata is only added for native eth too

Reply:
Also nit:

```
if (!Array.isArray(trades)) {
    trades = [ trades ]
}

return _.some(trades, trade => trade.priceImpact.greaterThan(..))
```

Reply:
> We need to check tokenIn is native eth too

doesn't this happen on line 363?

Reply:
Whoops, yes